File: binary_search/binarySearch.py
def binarySearch(array, target):
	left = 0
	right = len(array)-1
	while left <= right:
		middle = (left + right) // 2
		num = array[middle]
		if num == target:
			return middle
		elif target < num:
			right = middle - 1
		else:
			left = middle + 1
	return -1

File: binary_search/binarySearch.js
function binarySearch(array, target) {
	let left = 0,
			right = array.length-1;
	while (left <= right) {
		const middle = Math.floor((left + right) / 2);
		if (array[middle] === target) { return middle }
		console.log(array[middle], target)
		
		if (array[middle] < target) {
			left = middle + 1;
		} else {
			right = middle - 1;
		}
	}
	return -1
}
